tests/integration/pilot/multi_version_revision_test.go

```diff
@@ -149,15 +152,39 @@ func testAllEchoCalls(ctx framework.TestContext, echoInstances []echo.Instance)
 // installRevisionOrFail takes an Istio version and installs a revisioned control plane running that version
 // provided istio version must be present in tests/integration/pilot/testdata/upgrade for the installation to succeed
 func installRevisionOrFail(ctx framework.TestContext, version string, configs map[string]string) {
-	installationTestdataDir := filepath.Join(env.IstioSrc, "tests/integration/pilot/testdata/upgrade")
-	installationConfigPath := filepath.Join(installationTestdataDir, fmt.Sprintf("%s-install.yaml", version))
-	configBytes, err := ioutil.ReadFile(installationConfigPath)
+	config, err := ReadInstallFile(fmt.Sprintf("%s-install.yaml", version))
 	if err != nil {
-		ctx.Fatalf("could not read installation config at path %s: %v", installationConfigPath, err)
+		ctx.Fatalf("could not read installation config: %v", err)
 	}
+	configs[version] = config
+	ctx.Config().ApplyYAMLOrFail(ctx, i.Settings().SystemNamespace, config)
+}
 
-	configs[version] = string(configBytes)
-	ctx.Config().ApplyYAMLOrFail(ctx, i.Settings().SystemNamespace, string(configBytes))
+// ReadInstallFile reads a tar compress installation file from the embedded
```

Comment:
there's an operator/tgz package and it would be be great to combine this function with that and move it into a root pkg. it's just a "nice to have" so i'll approve with a hold and you can decide if you have time to do it.

Reply:
This is technically a tar not a tgz. But maybe it should be tgz... and use that